### DISCUSSION TITLE
Under cabinet LED that is white only with five different temperatures

### DISCUSSION THREAD

**Discussion by millerim** [make-all/tuya-local#2576] [Q&A] [answered]
Great work on Tuya Local!  I'm writing a device file for a Feit OneSync Under Cabinet LED that is white only, with five different temperatures from 2700K to 5000K, and four different brightness levels, no other color but white.  It's working in that I can turn the light on and off, select the temperature and the brightness, but I'm struggling to get the Home Assistant device to reflect that it is white only.  The device controls display a color picker and a set of favorite colors.  I'm certain my configuration is wrong, but I'm stumped.  I've attached the tinytuya output, the device yaml file, and a screenshot of the HA device.  Can you offer some coaching?  Thanks in advance!

[tinytuya.txt](https://github.com/user-attachments/files/17965992/tinytuya.txt)
![HA Device](https://github.com/user-attachments/assets/35e54ace-97d7-4c25-99e9-ac4437888628)

`name: OneSync Under Cabinet Smart Bridge
products:
  - id: md1jguhfrhdklqyh
    name: OneSync
primary_entity:
  entity: light
  dps:
    - id: 20
      type: boolean
      name: switch
    - id: 21
      type: string
      name: color_mode
      optional: true
      mapping:
        - dps_val: white
          value: white
        - dps_val: colour
          value: hs
        - dps_val: scene
          value: Scene
        - dps_val: null
          value: white
          hidden: true
    - id: 25
      type: json
      name: scene_data
      optional: true
    - id: 101
      name: brightness
      type: string
      mapping:
        - dps_val: "Low"
          value: 0
        - dps_val: "Mid"
          value: 82
        - dps_val: "High"
          value: 164
        - dps_val: "Max"
          value: 255
    - id: 102
      name: color_temp
      type: string
      mapping:
        - dps_val: 2700K
          value: 0
        - dps_val: 3000K
          value: 250
        - dps_val: 3500K
          value: 500
        - dps_val: 4000K
          value: 750
        - dps_val: 5000K
          value: 1000
secondary_entities:
  - entity: number
    name: timer
    category: config
    dps:
      - id: 26
        type: integer
        name: value
        optional: true
        unit: min
        range:
          min: 0
          max: 86400
        mapping:
          - scale: 60
            step: 60
  - entity: select
    icon: mdi:lightbulb-variant
    category: config
    dps:
      - id: 101
        type: string
        name: brightness
        optional: true
        mapping:
          - dps_val: Low
            value: Low
          - dps_val: Mid
            value: Mid
          - dps_val: High
            value: High
          - dps_val: Max
            value: Max
  - entity: select
    icon: mdi:lightbulb-variant
    name: color_temp
    category: config
    dps:
      - id: 102
        type: string
        name: option
        optional: true
        mapping:
          - dps_val: 2700K
            value: 2700K
          - dps_val: 3000K
            value: 3000K
          - dps_val: 3500K
            value: 3500K
          - dps_val: 4000K
            value: 4000K
          - dps_val: 5000K
            value: 5000K`

**Comment by make-all** [comment]
If you have "hs" as an option in the color_mode, it will be expected that the hue and saturation can be set in that mode.  Most lights just reuse that from the standard tuya lightbulb definition, but don't actually support it.  Try renaming the dp to something else (like work_mode) so the integration calculates a single supported color mode based on the dps that are available instead.

**Comment by millerim** [comment]
Thanks!  I changed that section to the following, and deleted and re-added the device but there was no change in HA.

    - id: 21
      type: string
      name: work_mode
      optional: true
      mapping:
        - dps_val: white
          value: White
        - dps_val: colour
          value: Color
        - dps_val: scene
          value: Scene
        - dps_val: music
          value: Music

**Comment by make-all** [comment, marked as answer]
Strange, there should be no color support any more, did you restart Home Assistant after changing the config?

But before you do that, the color_temp mappings should be to 2700, 3000, 3500, 4000, 5000 rather than values between 0 and 1000.

**Reply by millerim** [reply]
Brilliant, all sorted.  Here's the working code if you want to add to the list of supported devices.  Thanks!

`
name: OneSync Under Cabinet Smart Bridge
products:
  - id: md1jguhfrhdklqyh
    name: OneSync
primary_entity:
  entity: light
  dps:
    - id: 20
      type: boolean
      name: switch
    - id: 21
      type: string
      name: work_mode
      optional: true
      mapping:
        - dps_val: white
          value: White
        - dps_val: colour
          value: Color
        - dps_val: scene
          value: Scene
        - dps_val: music
          value: Music
    - id: 25
      type: json
      name: scene_data
      optional: true
    - id: 101
      name: brightness
      type: string
      mapping:
        - dps_val: "Low"
          value: 1
        - dps_val: "Mid"
          value: 82
        - dps_val: "High"
          value: 164
        - dps_val: "Max"
          value: 255
    - id: 102
      name: color_temp
      type: string
      mapping:
        - dps_val: 2700K
          value: 2700
        - dps_val: 3000K
          value: 3000
        - dps_val: 3500K
          value: 3500
        - dps_val: 4000K
          value: 4000
        - dps_val: 5000K
          value: 5000
secondary_entities:
  - entity: number
    name: timer
    category: config
    dps:
      - id: 26
        type: integer
        name: value
        optional: true
        unit: min
        range:
          min: 0
          max: 86400
        mapping:
          - scale: 60
            step: 60
  - entity: select
    icon: mdi:lightbulb-variant
    category: config
    dps:
      - id: 101
        type: string
        name: brightness
        optional: true
        mapping:
          - dps_val: Low
            value: Low
          - dps_val: Mid
            value: Mid
          - dps_val: High
            value: High
          - dps_val: Max
            value: Max
  - entity: select
    icon: mdi:lightbulb-variant
    name: color_temp
    category: config
    dps:
      - id: 102
        type: string
        name: option
        optional: true
        mapping:
          - dps_val: 2700K
            value: 2700K
          - dps_val: 3000K
            value: 3000K
          - dps_val: 3500K
            value: 3500K
          - dps_val: 4000K
            value: 4000K
          - dps_val: 5000K
            value: 5000K
`

**Reply by millerim** [reply]
Minor bug fixed with secondary brightness entity...

`
name: OneSync Under Cabinet Smart Bridge
products:
  - id: md1jguhfrhdklqyh
    name: OneSync
primary_entity:
  entity: light
  dps:
    - id: 20
      type: boolean
      name: switch
    - id: 21
      type: string
      name: work_mode
      optional: true
      mapping:
        - dps_val: white
          value: White
        - dps_val: colour
          value: Color
        - dps_val: scene
          value: Scene
        - dps_val: music
          value: Music
    - id: 25
      type: json
      name: scene_data
      optional: true
    - id: 101
      name: brightness
      type: string
      mapping:
        - dps_val: "Low"
          value: 1
        - dps_val: "Mid"
          value: 82
        - dps_val: "High"
          value: 164
        - dps_val: "Max"
          value: 255
    - id: 102
      name: color_temp
      type: string
      mapping:
        - dps_val: 2700K
          value: 2700
        - dps_val: 3000K
          value: 3000
        - dps_val: 3500K
          value: 3500
        - dps_val: 4000K
          value: 4000
        - dps_val: 5000K
          value: 5000
secondary_entities:
  - entity: number
    name: timer
    category: config
    dps:
      - id: 26
        type: integer
        name: value
        optional: true
        unit: min
        range:
          min: 0
          max: 86400
        mapping:
          - scale: 60
            step: 60
  - entity: select
    icon: mdi:lightbulb-variant
    name: brightness
    category: config
    dps:
      - id: 101
        type: string
        name: option
        optional: true
        mapping:
          - dps_val: Low
            value: Low
          - dps_val: Mid
            value: Mid
          - dps_val: High
            value: High
          - dps_val: Max
            value: Max
  - entity: select
    icon: mdi:lightbulb-variant
    name: color_temp
    category: config
    dps:
      - id: 102
        type: string
        name: option
        optional: true
        mapping:
          - dps_val: 2700K
            value: 2700K
          - dps_val: 3000K
            value: 3000K
          - dps_val: 3500K
            value: 3500K
          - dps_val: 4000K
            value: 4000K
          - dps_val: 5000K
            value: 5000K
`
